src/utils/MediaEventHelper.ts

```diff
@@ -26,22 +26,46 @@ import { IDestroyable } from "./IDestroyable";
 
 // TODO: We should consider caching the blobs. https://github.com/vector-im/element-web/issues/17192
 
+interface ITypedBlob {
+    mimetype: string;
+    data: Blob;
+}
+
+// infer type of blob from blob itself
+function toTypedBlob(blob: Blob): ITypedBlob {
+    return {
+        mimetype: blob.type,
+        data: blob,
+    };
+}
+
+async function createDataUrl(blob: ITypedBlob): Promise<string> {
+    return `data:${blob.mimetype};base64,${await blob.data.text().then(btoa)}`;
+}
+
 export class MediaEventHelper implements IDestroyable {
     // Either an HTTP or Object URL (when encrypted) to the media.
     public readonly sourceUrl: LazyValue<string>;
     public readonly thumbnailUrl: LazyValue<string>;
 
     // Either the raw or decrypted (when encrypted) contents of the file.
-    public readonly sourceBlob: LazyValue<Blob>;
-    public readonly thumbnailBlob: LazyValue<Blob>;
```

Comment:
these were deliberately exposed and should not be hidden. It's for overall performance of the app.

Reply:
(also used by projects downstream of us)